src/pytorch_lightning/strategies/launchers/multiprocessing.py

```diff
@@ -153,7 +153,7 @@ def _recover_results_in_main_process(self, worker_output: "_WorkerOutput", train
         # load last weights
         if worker_output.weights_path is not None:
             ckpt = self._strategy.checkpoint_io.load_checkpoint(worker_output.weights_path)
-            trainer.lightning_module.load_state_dict(ckpt)
+            trainer.lightning_module.load_state_dict(ckpt, strict=False)
```

Comment:
(not for this PR). Is there a way we could know which modules were not loaded? so that we tell the user about this. Otherwise it could be confusing if they notice part is not loaded.

Reply:
Yes there is. See the earlier commit where I had this: https://github.com/Lightning-AI/lightning/pull/16365/commits/cefa66e21c6bcced4ed69e0c1b01fecbf966350c

But the issue is with us know knowing whether anything meaningful happens after `trainer.fit()` where the user relies on the full model. If not, this would just be a false positive warning. 


Case 1:
```py
trainer.fit(model)
# program exits
```
All good. No warnings needed.

Case 2:
```py
trainer.fit(model)
trainer.test(model, ckpt_path="best")
# or
trainer.test()  # no model instance given
```
All good. setup() gets called and all weights reloaded for test. No warnings needed.

Case 3:
```py
# or
trainer.test(model)  # use the current weights (no ckpt_path=...)
# when setup() gets called, the model weights would get overwritten
```
A warning may or may not be appropriate. 

Case 4:
```py
trainer.fit(model)

# do something with the model later
model.forward(...)  # some layers are not available, this would error 
```
A warning may or may not be appropriate. The warning would provide additional information as to why the error occurs.

Reply:
I prefer a false positive (we have `PossibleUserWarning` for this) than Case 3 creating a footgun for our users.

Reply:
I tend to agree with @awaelchli here. I think a prominent warning in the docs should be sufficient. Let's not spam warnings with inherent properties of our strategies